nimble/host/src/ble_l2cap_sig.c

```diff
@@ -1979,6 +1979,9 @@ ble_l2cap_sig_ping(uint16_t conn_handle, ble_l2cap_ping_fn cb,
 
 
     rc = ble_l2cap_sig_tx(proc->conn_handle, txom);
+    if (rc != 0) {
```

Comment:
hmm I think ble_l2cap_sig_tx is (or should) consume txom regardless of result

Reply:
Nope, unfortunately it does not. I have been debugging some of my nodes stopping to scan properly, until I found out that the node's msys buffer simply run out of mbufs, caused by this function not freeing `txom` on error...

Reply:
For further verification, here the code of `ble_l2cap_sig_tx()`:
```c
int
ble_l2cap_sig_tx(uint16_t conn_handle, struct os_mbuf *txom)
{
    struct ble_l2cap_chan *chan;
    struct ble_hs_conn *conn;
    int rc;

    ble_hs_lock();
    rc = ble_hs_misc_conn_chan_find_reqd(conn_handle, BLE_L2CAP_CID_SIG,
                                         &conn, &chan);
    if (rc == 0) {
        rc = ble_l2cap_tx(conn, chan, txom);
    }
    ble_hs_unlock();

    return rc;
}
```
As one can see, no freeing of `txom` in case `ble_hs_misc_conn_chan_find_reqd()` fails...

Reply:
hmm but then, ble_l2cap_tx will consume txom even on error
maybe ble_l2cap_sig_tx() should be updated to always consume txom,  btw I think ble_sm_tx() has same issue

I'll try to lo review those code parts this week and propose some unified solution

Reply:
I completely agree, this makes sense to be fixed coherently. Debugging this kind of inconsistency is no fun :-)

Let me know where I can help!